### PR TITLE
Bootstrap CLI to hyper startup

### PR DESCRIPTION
**Pull request by jaydh**
This bootstraps the CLI into the hyper startup process. Namely, this bypasses cross-platform installation for a separate binary that has caused issues for linux users. I've outlined some of the details [here](https://github.com/zeit/hyper/issues/2998).

The main CLI function returns a flag for the main thread to execute the terminal emulator if and only if hyper is run without any command line arguments (both in dev and prod). If there are arguments the CLI parses them, executing the commands, or showing a help screen and terminating the node process altogether.

### CODE REVIEW

File: app/cli/index.js [hunk 1]
```diff
@@ -82,15 +88,18 @@ const lsRemote = pattern => {
...
+    .finally(() => process.exit(0));
 };
 
 args.command(['s', 'search'], 'Search for plugins on npm', (name, args_) => {
+  commandInvoked = true;
   const spinner = ora('Searching').start();
   const query = args_[0] ? args_[0].toLowerCase() : '';
-
-  commandPromise = lsRemote(query)
+  console.log(1);
```
Comment (nazariyv) [inline app/cli/index.js]: probably don't need this here?

Comment (jaydh) [reply app/cli/index.js]: Yup good catch! There's also a separate log statement on 102 we can delete

File: app/cli/index.js [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 const api = require('./api');
-
+const isDev = require('electron-is-dev');
+const promiseFinally = require('promise.prototype.finally');
+promiseFinally.shim();
 const PLUGIN_PREFIX = 'hyper-';
 
-let commandPromise;
-
+const configLocation = process.platform === 'win32' ? process.env.userprofile + '\\.hyper.js' : '~/.hyper.js';
```
Comment (ppot) [inline app/cli/index.js]: Will crash with new location of `.hyper.js`

File: app/cli/index.js (hunk 1 above)
Comment (ppot) [inline app/cli/index.js]: remove

File: app/cli/index.js [hunk 3]
```diff
@@ -82,15 +88,18 @@ const lsRemote = pattern => {
...
 args.command(['s', 'search'], 'Search for plugins on npm', (name, args_) => {
+  commandInvoked = true;
   const spinner = ora('Searching').start();
   const query = args_[0] ? args_[0].toLowerCase() : '';
-
-  commandPromise = lsRemote(query)
+  console.log(1);
+  lsRemote(query)
     .then(entries => {
+      console.log(2);
```
Comment (ppot) [inline app/cli/index.js]: remove

**Comment by Stanzilla** [comment]
Hey @jaydh are you willing to pick this up again?

File: app/cli/index.js (hunk 2 above)
Comment (Stanzilla) [reply app/cli/index.js]: @jaydh

**Pull request closed without merging**